Formula/llvm@11.rb

```diff
@@ -276,10 +299,69 @@ def caveats
     system "#{bin}/clang++", "-v",
            "-isystem", "#{opt_include}/c++/v1",
            "-std=c++11", "-stdlib=libc++", "test.cpp", "-o", "testlibc++",
-           "-L#{opt_lib}", "-Wl,-rpath,#{opt_lib}"
-    on_macos { assert_includes MachO::Tools.dylibs("testlibc++"), "#{opt_lib}/libc++.1.dylib" }
+           "-rtlib=compiler-rt", "-L#{opt_lib}", "-Wl,-rpath,#{opt_lib}"
+    assert_includes (testpath/"testlibc++").dynamically_linked_libraries,
+      (opt_lib/shared_library("libc++", "1")).to_s
+    (testpath/"testlibc++").dynamically_linked_libraries.each do |lib|
+      refute_match(/libstdc\+\+/, lib)
+      refute_match(/libgcc/, lib)
+      refute_match(/libatomic/, lib)
+    end
     assert_equal "Hello World!", shell_output("./testlibc++").chomp
```

Comment:
This is where the test is failing. I think the new flags you added to the compilation of `testlibc++` aren't as innocuous as we thought.

Reply:
I have had trouble with Compiler-RT on ARM in a separate PR, so it's possible it just doesn't work properly, and adding `-rtlib=compiler-rt` has exposed that problem.

Reply:
I'm a little confused though, because I had a previous that was successful with that flag: https://github.com/Homebrew/homebrew-core/actions/runs/948629092.  The one thing I did change was that I used the `install/strip` target.  If that ends up creating a problem on ARM for `compiler-rt`, then that could indicate that `install/strip` breaks `compiler-rt` on ARM.

Reply:
Indeed, going back to `install/strip` has fixed the test on ARM.  I am thinking we should not use `install/strip` on ARM for LLVM 11, as `compiler-rt` may not be the only thing it breaks.  I think LLVM 11 was the first version to support Apple Silicon, so perhaps the symbol stripping for that architecture had some problems which were resolved in LLVM 12?

Reply:
Hmmm. Very weird. I suspect we'll want to avoid `install/strip` on LLVM 12 too. We can just go back to the `on_linux` block you wanted to do originally.

Reply:
What is the size difference from stripping on Linux?

Reply:
I haven't built an unstripped version on Linux before, but that's a good question.  I'll build an unstripped bottle for LLVM 12 and see what that looks like in terms of size.  Unless the size saving in Linux is substantial, I'm not sure it's worth the risk of breakage.  

If I hadn't added `-rtlib=compiler-rt` to that test, we probably would never have known that stripping can break `compiler-rt`.  Even if that particular issue is specific to ARM on LLVM 11, who knows what else is broken?

Reply:
The stripped bottle for LLVM 12 is 298MB. The unstripped bottle 316MB.  I'm not really sure that saving in size is worth the risk of breakage from stripping.  Does @Homebrew/linux have any opinions on this?

Reply:
How do the untarred sizes compare?

Reply:
The unstripped untarred folder is 1.4GB, while the stripped folder is 1.3GB.  On the level of individual files, `libLLVM.so` is 115.6 MB unstripped and 98.8 MB stripped, and the differences get smaller from there.  We're probably talking about <10% reduction in file size overall.

Reply:
Probs not worth it, given that we don't have a clear idea of what it does and doesn't break. Especially given how we use `LLVM_ENABLE_RUNTIMES`, which means we have a bunch of things that _aren't_ built with our shims and hence may contain things like debug symbols, etc.

Reply:
I think that back then the difference between stripped and unstripped was bigger. And that I was manually building llvm and manually uploading it with a slow connection, like 10 to 20 times a year. One upload took between 1 and 2 hours, so every megabyte I could squeeze out of that build was worth it.

I am fine to remove the stripping.

Reply:
It might be that those were `Debug` builds (cf. https://github.com/Homebrew/brew/issues/11455), which is why stripping made a big difference.

Reply:
Yeah if were unintentionally making them debug builds, removing the debugging symbols would make a much bigger difference than removing unused symbols.  If we're not doing that anymore, that could probably explain why the effect is so small.

@carlocab with this resolved, I think we are ready to merge and nothing should need to be rebottled on macOS.  I'll have to open a new PR to disable stripping in LLVM 12, and I won't add it for the older LLVM versions either.

@iMichka I think LLVM 11 and 12 should be revision bumped to disable stripping. Can that be handled in the merge?  Or do we need to open new PRs for that?  I'll change the PRs I opened for LLVM 7, 8, and 9 to disable stripping on those versions as well.

Reply:
I see @carlocab already took of LLVM 12, so that I've crossed that out.